app/assets/javascripts/discourse/app/initializers/category-color-css-generator.js

```diff
@@ -0,0 +1,31 @@
+export default {
+  name: "category-color-css-generator",
+  after: "register-hashtag-types",
+
+  /**
+   * This generates CSS variables for each category color,
+   * which can be used in themes to style category-specific elements.
+   *
+   * It is also used when styling hashtag icons, since they are colored
+   * based on the category color.
+   */
+  initialize(container) {
+    this.site = container.lookup("service:site");
+
+    let generatedCssVariables = [":root {"];
+
+    this.site.categories.forEach((category) => {
+      generatedCssVariables.push(
+        `--category-${category.id}-color: #${category.color};`
+      );
+    });
+
+    generatedCssVariables.push("}");
```

Comment:
🏌️ ⛳  (you don't have to use this, I just can't help myself 😂)

```suggestion
    const generatedCssVariables = [
      ":root {",
      ...this.site.categories.map((category) =>
        `--category-${category.id}-color: #${category.color};`
      ),
      "}"
    ];
```

Reply:
That's some nice golfing Mr. CvX

![image](https://user-images.githubusercontent.com/920448/229961098-bbc5e15c-fe8a-4223-86e6-6d012525431b.png)

Will use this :)